### ISSUE TITLE
Use libieee1284 instead of native port-mapped I/O

### ISSUE THREAD

**Issue by volkertb** [yeokm1/covox-music-player#2]
Hi. The current code is platform- and architecture-dependent, since it uses x86-style port-mapped I/O to send the bytes to the parallel port.

By using something like [libieee1284](https://github.com/twaugh/libieee1284) instead, the code can be made portable, so people  can use this software to enjoy their Covox devices on other operating systems and/or architectures as well. :relaxed:

Also, it wouldn're require the covox-music-player to always have to run with sudo privileges, as long as the current user has access to the parallel port interface.

**Comment by volkertb** [comment]
@yeokm1 Well, I tested this change in a QEMU VM with an emulated parallel port. It seemed to work, although it did still require sudo privileges. :sweat_smile: Instead of an x86-specific I/O port, the device name was `parport0`.

See the screenshot below.

I did need to run it with sudo privileges to output anything to it, though. Also, I'm not sure if the audio is actually working, since the emulated parallel port wasn't routed to an actual physical parallel port on the host. Also, the number of skipped frames is quite high. But that might have something to do with the emulated parallel port and my (failed) attempt to have QEMU dump the output of the emulated parallel port to a file. Testing this with actual hardware is better, of course.

I currently don't have an LPT DAC to test this with. I'll probably order a CVX4, the Covox clone I mentioned in my addition to the README. And I have another computer with a physical parallel port that I can test this with.

Also, I might soon be able to test this with a Ultrasparc workstation as well. That way, we'll know for sure whether the code is cross-platform with this change. :smile:

Oh, and libieee1284 is also supposed to be cross-platform in terms of operating systems, not just CPU architecture. It should therefore make it easier to port this project to other operating systems in the future, should you be interested in that.

![Screenshot from 2020-10-11 02-50-22](https://user-images.githubusercontent.com/518023/95668004-f4b86c80-0b6d-11eb-9d4c-956b2adddca5.png)

**Comment by yeokm1** [comment]
@volkertb Thanks so much for your effort! I honestly didn't expect anyone to be using this project. 

At that time when I started this project, I don't think the CVX4 was available yet. I agree with you that is good way to have an off-the-shelf Covox experience. 

I'll need some time to review your request and test the code. I have to fire up my retro PC for a start! I have a CVX4 as well so I'll test with that too.

I don't have a non-x86 system with a parallel port though to test the cross-platform aspects but I believe it should work.

**Comment by yeokm1** [comment]
Hi @volkertb 

Life stuff cropped up and this issue slipped my mind.

I merged your PR to a [`ieee-test`](https://github.com/yeokm1/covox-music-player/tree/ieee-test) branch and tried to run it. But I got an error I'm not sure where the source of it is.

I'm using Kubuntu 22.04.3 LTS with kernel 6.2.0-26-generic with a Gigabyte H310M S2P 2.0 motherboard which still has a native parallel port. I have also tested with a Thinkpad T42 with Debian 12 kernel 6.1.0-10-686 with same result.

```bash
root@...  ./covox-music-player-linux.out parport0 test.wav
...
Parallel ports detected on this system:
 * Address/name: parport0 (selected)
...
Attempting to open parallel port at parport0
Could not claim parallel port parport0.Do you have root privileges? IEEE1284: Error interfacing system
```

I tried printing the errno code and it shows 6 which refers to `ENXIO 6 No such device or address` when using `errno -l`.

```bash
root@... ./covox-music-player-linux.out parport0 file.wav
...
Parallel ports detected on this system:
 * Name: parport0, Address: 888 (selected)

Attempting to open parallel port at parport0
after claim errno code 6
Could not claim parallel port parport0.Do you have root privileges? IEEE1284: Error interfacing system
```

The base address also shows `888` which is odd. This version is committed to this branch so you can try it.

 The existing code still works with address 0x378 directly on both systems.

**Comment by volkertb** [comment]
> Life stuff cropped up and this issue slipped my mind.

Yeah, I know what you mean. :sweat_smile: Sorry for the late reply. Happy New Year! :star2:

I'll try to find some time to look into this, since I'd like to finish this work and get it merged.

Once we've verified this to work in an x86 system with parallel ports at the legacy I/O addresses, let me test this PR on a different architecture as well.

I have a Sun Blade 150 (UltraSPARC IIi) workstation that I believe has a PC-style (Centronics compatible) parallel port, which would be a good test subject for this. :slightly_smiling_face:

**Comment by yeokm1** [comment]
Alright, I'll wait for your test.
